src/bloqade/qubit/_prelude.py

```diff
@@ -0,0 +1,43 @@
+from typing import Annotated
+
+from kirin import ir
+from kirin.passes import Default
+from kirin.prelude import structural_no_opt
+from typing_extensions import Doc
+
+from . import _dialect as qubit
+
+
+@ir.dialect_group(structural_no_opt.union([qubit]))
+def kernel(self):
+    """Compile to a qubit kernel"""
+
+    def run_pass(
+        mt,
+        *,
+        verify: Annotated[
+            bool, Doc("run `verify` before running passes, default is `True`")
+        ] = True,
+        typeinfer: Annotated[
+            bool,
+            Doc(
+                "run type inference and apply the inferred type to IR, default `False`"
+            ),
+        ] = False,
+        fold: Annotated[bool, Doc("run folding passes")] = True,
+        aggressive: Annotated[
+            bool, Doc("run aggressive folding passes if `fold=True`")
+        ] = False,
+        no_raise: Annotated[bool, Doc("do not raise exception during analysis")] = True,
```

Comment:
I assume for the previous annotations if you mention the default value, you should carry that pattern over here too?